### DISCUSSION TITLE
using-serilog-aspnetcore-in-asp-net-core-3-reducing-log-verbosity/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#320] [Announcements]
# using-serilog-aspnetcore-in-asp-net-core-3-reducing-log-verbosity/

In this post I describe how you can use Serilog.AspNetCore to reduce the number of logs generated by every request in ASP.NET Core 3.0

https://andrewlock.net/using-serilog-aspnetcore-in-asp-net-core-3-reducing-log-verbosity/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at January 23 2020, 08:52, in reply to [Kaushik Roy Chowdhury](https://disqus.com/by/kaushik_roy_chowdhury/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/320#discussioncomment-6058026)</em>

---

<p>Of course, no problem, I'd just appreciate it if you mention the blog :)</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at February 18 2020, 08:42, in reply to [Kaushik Roy Chowdhury](https://disqus.com/by/kaushik_roy_chowdhury/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/320#discussioncomment-6058032)</em>

---

<p>Well, that's interesting! It looks like Microsoft have changed the default log levels in the Development environment from all project templates! Not sure if that was in 3.1 or a patch release, but it means the same log levels are used in both development and production:</p><p>Previously, log levels in appsettings.Development.json were this:</p><p></p><pre><code><br>{<br>  "Logging": {<br>    "LogLevel": {<br>      "Default": "Debug",<br>      "System": "Information",<br>      "Microsoft": "Information"<br>    }<br>  }<br>}<br></code></pre><p></p><p>Now they're this:</p><p></p><pre><code><br>{<br>  "Logging": {<br>    "LogLevel": {<br>      "Default": "Information",<br>      "Microsoft": "Warning",<br>      "Microsoft.Hosting.Lifetime": "Information"<br>    }<br>  }<br>}<br></code></pre><p></p><p>You can see the change here: <a href="https://github.com/dotnet/aspnetcore/pull/14748/files" rel="nofollow noopener" title="https://github.com/dotnet/aspnetcore/pull/14748/files">https://github.com/dotnet/aspnetcore/pull/14748/files</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 09 2020, 10:32, in reply to [wswieboda](https://disqus.com/by/wswieboda/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/320#discussioncomment-6058039)</em>

---

<p>Yes, that assumes you don't want to use the endpoint routing features though, using the "old" map options instead of the 3.x approach.</p><p>That's fine, and honestly is probably easier in many cases, but is no good if you want to use endpoint routing features like authorization on your health checks, then that approach won't work</p>

**Comment by dlwiii** [comment]
How would I get rid of the request messages like this:

[14:29:53 INF] HTTP GET / responded 200 in 129.9509 ms

Don't agonize about why, I just want to know how.
